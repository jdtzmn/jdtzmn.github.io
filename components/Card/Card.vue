<template lang="pug">
  .card
    slot
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.component('card', {})
</script>

<style lang="scss" scoped>
@import '~@/styles/colors';
@import '~@/styles/font';

.card {
  // padding
  padding: 0.5em 0;
}

.card > .card-header {
  @extend %text-heading;

  // font
  font-size: 1.2em;

  // border
  border-bottom: 1px solid $lightgray;
}
</style>
