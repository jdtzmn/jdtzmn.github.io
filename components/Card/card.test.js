import { shallowMount } from '@vue/test-utils'
import Card from './Card'

describe('Card', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(Card, {
      slots: {
        default: '<div class="slot"></div>'
      }
    })
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render card div', () => {
    expect(wrapper.contains('.card')).toBeTruthy()
  })

  it('slot should render inside card div', () => {
    expect(wrapper
      .find('.card')
      .contains('.slot'))
      .toBeTruthy()
  })
})
