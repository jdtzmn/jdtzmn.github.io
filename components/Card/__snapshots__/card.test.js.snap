// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div class="card">
  <div class="slot"></div>
</div>
`;
